src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -231,14 +233,25 @@ protected Object readResolve() {
     	BuildVariableResolver resolver = build instanceof AbstractBuild ? new BuildVariableResolver((AbstractBuild<?,?>)build, listener) : null;
 		Map<String, String> properties = m_scanner.getProperties(resolver);
 		properties.put(CoreConstants.SCANNER_TYPE, m_scanner.getType());
-                properties.put(CoreConstants.APP_ID,  m_application);
+        properties.put(CoreConstants.APP_ID,  m_application);
 		properties.put(CoreConstants.SCAN_NAME, m_name + "_" + SystemUtil.getTimeStamp()); //$NON-NLS-1$
 		properties.put(CoreConstants.EMAIL_NOTIFICATION, Boolean.toString(m_emailNotification));
 		properties.put("APPSCAN_IRGEN_CLIENT", "Jenkins-" + SystemUtil.getOS());
 		properties.put("ClientType", "Jenkins-" + SystemUtil.getOS());
+		properties.put("APPSCAN_CLIENT_VERSION", Jenkins.VERSION);
+		properties.put("IRGEN_CLIENT_PLUGIN_VERSION", getPluginVersion());
 		return properties;
     }
     
+    private String getPluginVersion() {
+    	Plugin tempPlugin = Jenkins.getInstance().getPlugin("appscan");
+	
+    	if(tempPlugin.getWrapper().getVersion() != null) {
+    		return tempPlugin.getWrapper().getVersion();
+    	}
+		
+    	return "";
```

Comment:
Wasn't sure if there was something we wanted to return in case there was some sort of error?  I can add it if needed?